File: api-asn/csharp/src/api/asn/esya/EESYACardSerialNumber.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using tr.gov.tubitak.uekae.esya.api.asn;
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.asn.esya;

namespace tr.gov.tubitak.uekae.esya.api.asn.esya
{
    public class EESYACardSerialNumber : BaseASNWrapper<ESYACardSerialNumber>, ExtensionType
    {
        public EESYACardSerialNumber(byte[] aBytes)
            : base(aBytes, new ESYACardSerialNumber()){}

        public EESYACardSerialNumber(String value)
            : base(new ESYACardSerialNumber(value)){}

        public EESYACardSerialNumber(ESYACardSerialNumber aObject)
            : base(aObject){}

        public EExtension toExtension(bool aCritic) {           
             return new EExtension(EESYAOID.oid_ESYA_CardSerialNumber, aCritic, getBytes());
        }
    }
}

File: api-asn/src/tr/gov/tubitak/uekae/esya/api/asn/esya/EESYACardSerialNumber.java
package tr.gov.tubitak.uekae.esya.api.asn.esya;

import com.objsys.asn1j.runtime.Asn1ValueParseException;
import tr.gov.tubitak.uekae.esya.api.asn.BaseASNWrapper;
import tr.gov.tubitak.uekae.esya.api.asn.x509.EExtension;
import tr.gov.tubitak.uekae.esya.api.asn.x509.ExtensionType;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.asn.esya.ESYACardSerialNumber;


public class EESYACardSerialNumber extends BaseASNWrapper<ESYACardSerialNumber> implements ExtensionType {
	

    public EESYACardSerialNumber(byte[] aBytes) throws ESYAException {
        super(aBytes, new ESYACardSerialNumber());
    }

    public EESYACardSerialNumber(String value) throws Asn1ValueParseException {
        super(new ESYACardSerialNumber(value));
    }

    public EESYACardSerialNumber(ESYACardSerialNumber aObject) {
        super(aObject);
    }

    public EExtension toExtension(boolean aCritic) throws ESYAException {
		return new EExtension(EESYAOID.oid_ESYA_CardSerialNumber, aCritic, this);
	}

}
